libtiledbsoma/src/soma/managed_query.cc

```diff
@@ -1231,11 +1237,13 @@ ManagedQuery::_extend_and_evolve_schema_with_details<std::string>(
     if ((strcmp(value_schema->format, "U") == 0) ||
         (strcmp(value_schema->format, "Z") == 0)) {
         uint64_t* offsets = (uint64_t*)value_array->buffers[1];
-        offsets_v.assign(offsets, offsets + num_elems + 1);
+        offsets_v.assign(
+            offsets + value_array->offset,
+            offsets + value_array->offset + num_elems + 1);
```

Comment:
Apologies if I am missing something obvious -- why is there a `+ 1` at the end of the second argument?  The `assign` uses normal iterator ranges [first, last), so I believe it should just be `offsets + value_array->offset + num_elems`

Reply:
@bkmartinjr that predates this PR. I'll look into it. Thanks for flagging.

Reply:
@bkmartinjr if the inputs are `["hello", "goodbye"]` then the first offset is 0, the second is 5, and the third 12. This makes it possible to locate the end of the string `"goodbye"` within the char buffer `"hellogoodbye"`.

Reply:
https://github.com/single-cell-data/TileDB-SOMA/pull/3815/commits/301a3b503cb8c3010ee86118c3fb362b3292c674

Reply:
More generally, it takes n+1 offsets to locate the starts and ends of n values in a string column. Please also see https://github.com/single-cell-data/TileDB-SOMA/pull/3815/commits/1ee11a25c0c961e39f7d14cc904934f899e12599.